src/main/java/org/sagebionetworks/bridge/udd/helper/ZipHelper.java

```diff
@@ -0,0 +1,52 @@
+package org.sagebionetworks.bridge.udd.helper;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.ByteStreams;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/** This helper zips the given input files into the given target file. */
+@Component
+public class ZipHelper {
+    private FileHelper fileHelper;
+
+    /** File helper, used to read data from the input files and write to the output file. */
+    @Autowired
+    public final void setFileHelper(FileHelper fileHelper) {
+        this.fileHelper = fileHelper;
+    }
+
+    /**
+     * Zips the list of input files and writes the result to the output file.
+     *
+     * @param fromList
+     *         list of input files
+     * @param to
+     *         output file to write the zip file to
+     * @throws IOException
+     *         if reading from input or writing to output fails
+     */
+    public void zip(List<File> fromList, File to) throws IOException {
+        try (BufferedOutputStream bufferedOutputStream = new BufferedOutputStream(fileHelper.getOutputStream(to));
```

Comment:
By convention, close() always calls flush() (and try-with-resources always calls close). While this might not be true in the general case, I've tested the whole downloader manually and have confirmed that this properly writes the zip file.